src/cryptonote_core/cryptonote_tx_utils.cpp

```diff
@@ -320,10 +351,17 @@ namespace cryptonote
       else
       {
         // sending to the recipient; derivation = r*A (or s*C in the subaddress scheme)
-        r = crypto::generate_key_derivation(dst_entr.addr.m_view_public_key, txkey.sec, derivation);
+        r = crypto::generate_key_derivation(dst_entr.addr.m_view_public_key, need_additional_txkey ? additional_txkey.sec : txkey.sec, derivation);
       }
       CHECK_AND_ASSERT_MES(r, false, "at creation outs: failed to generate_key_derivation(" << dst_entr.addr.m_view_public_key << ", " << txkey.sec << ")");
 
+      if (need_additional_txkey || (!dont_need_additional_txkey && crypto::rand<size_t>() % 100 < PROBABILITY_MAKE_DUMMY_ADDITIONAL_TX_PUBKEY * 100))
+      {
+        additional_tx_public_keys[output_index] = additional_txkey.pub;
+        additional_tx_keys[output_index] = additional_txkey.sec;
+        LOG_PRINT_L2("We made an additional txpubkey for output index " << output_index << (need_additional_txkey ? "" : " which isn't necessary for receiving but useful for obfuscation"));
+      }
```

Comment:
No, that's not the case. When there are two outputs (i.e. one change and one destination), `dont_need_additional_txkey` would be `true` because `(num_stdaddresses, num_subaddresses)` would be either `(0, 1)` or `(1, 0)`, see L331.